src/execution/mod.rs

```diff
@@ -68,6 +71,29 @@ where
         Ok(instance)
     }
 
+    pub fn invoke_named<Param: InteropValueList, Returns: InteropValueList>(
+        &mut self,
+        func_name: &str,
+        param: Param,
+    ) -> Result<Returns> {
+        let func_idx = self.exports.iter().find_map(|export| {
```

Comment:
This is a linear search through all registered functions. That is fine for now, but will require optimization later. Please add a `// TODO optimize function search` just for us to pick it up eventually, please.